File: D_Nested_Segments.py
import bisect

INF = 0x3f3f3f3f
mod = 10**9 + 7
N = 4 * 10**5 + 1

# 离散化数组
b = []

# 更新树
tree = [0] * N

def lowbit(x):
    return x & -x

def idx(x):
    return bisect.bisect_left(b, x)

def update(i, x):
    while i < N:
        tree[i] += x
        i += lowbit(i)

def query(i):
    ans = 0
    while i > 0:
        ans += tree[i]
        i -= lowbit(i)
    return ans

n = int(input())
a = []
ans = [0 for _ in range(n)]

for i in range(n):
    x, y = map(int, input().split())
    a.append((i, (x, y)))
    b.append(x)
    b.append(y)

a.sort(key=lambda x: x[1][0])
b.sort()
b = list(dict.fromkeys(b))  # 去重

for i in range(n):
    update(idx(a[i][1][1]), 1)

for i in range(n):
    update(idx(a[i][1][1]), -1)
    ans[a[i][0]] = query(idx(a[i][1][1]))

for x in ans:
    print(x)

File: D_Nested_Segments.cpp
#include <bits/stdc++.h>
using namespace std;

#define ff first
#define ss second
#define int long long
#define lowbit(x) x & -x

const int INF = 0x3f3f3f3f;
const int mod = 1e9 + 7; // 998244353;
const int N = 4e5 + 1;

vector<int> b; // 离散化数组

int idx(int x) {
    return lower_bound(b.begin(), b.end(), x) - b.begin();
}

int tree[N];
void update(int i, int x) {
    while (i <= N) {
        tree[i] += x;
        i += lowbit(i);
    }
}

int query(int i) {
    int ans = 0;
    while (i) {
        ans += tree[i];
        i -= lowbit(i);
    }
    return ans;
}

signed main() {
    cin.tie(nullptr)->sync_with_stdio(false);
    cout << fixed << setprecision(15);
    
    int n;
    cin >> n;
    vector<pair<int, pair<int, int>>> a(n);
    vector<int> ans(n);
    for (int i = 0; i < n; ++i) {
        a[i].ff = i;
        cin >> a[i].ss.ff >> a[i].ss.ss;
        b.emplace_back(a[i].ss.ff);
        b.emplace_back(a[i].ss.ss);
    }
    ranges::sort(a, [](pair<int, pair<int, int>> x, pair<int, pair<int, int>> y) {
        return x.ss.ff < y.ss.ff;
    });
    ranges::sort(b);
    b.erase(unique(b.begin() ,b.end()), b.end());
    for (int i = 0; i < n; ++i) {
        update(idx(a[i].ss.ss), 1);
    }
    for (int i = 0; i < n; ++i) {
        update(idx(a[i].ss.ss), -1);
        ans[a[i].ff] = query(idx(a[i].ss.ss));
    }
    for (int i = 0; i < n; ++i) {
        cout << ans[i] << endl;
    }
    
    return 0;
}
